### PR TITLE
[kops] Add targets and README.md

### PR DESCRIPTION
**Pull request by osterman**
## what
* Add a `README.md` to roughly describe the process
* Add some **simple** targets that are common when provisioning a new cluster

## why
* The `Makefile` targets serve as *executable documentation*; that is, they can be copied & pasted just as easily as they can be run. They are not difficult to understand.
* This makes writing the documentation easier that pasting commands into the `README.md` which will just grow stale.

### CODE REVIEW

File: aws/kops/main.tf [hunk 1]
```diff
@@ -144,6 +144,14 @@ resource "aws_ssm_parameter" "kops_utility_subnets" {
   overwrite   = "true"
 }
 
+resource "aws_ssm_parameter" "kops_non_masquerade_cidr" {
+  name        = "${format(var.chamber_parameter_name, local.chamber_service, "kops_non_masquerade_cidr")}"
+  value       = "${var.kops_non_masquerade_cidr}"
+  description = "Kops utility subnet CIDRs"
```
Comment (aknysh) [inline aws/kops/main.tf]:
```suggestion
  description = "The CIDR range for Pod IPs"
```

File: aws/kops/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+5. Run `make kops/build-manifest` to compile the configuration template with current environment settings
+6. Run `make kops/create` to submit the cluster state manifest. Note, no resources will be provisioned.
+7. Run `make kops/create-secret-sshpublickey` to provision the SSH public key. Note, the public key was created in the `make apply` step and requires `/secrets/tf` to be mounted. Mount this directory by running `mount -a`.
+8. Run `make kops/plan` to view the proposed cluster
+9. Run `make kops/apply` to build the cluster
+10. Run `make kops/validate` to view cluster status. Note, it will take ~10 minutes to come online (depending on cluster size)
+
+Once the cluster is online, you can interact with it using `kubectl`. 
+
+To start, first run this to get the credentials:
```
Comment (aknysh) [inline aws/kops/README.md]:
```suggestion
To start, first run this to export `kubecfg` from the `kops` state store (required to access the cluster):
```

File: aws/kops/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+## Quick Start
+
+This is roughly the process to get up and running. These instructions assume you're running inside of a [Geodesic shell](https://github.com/cloudposse/geodesic).
+1. Update the `terraform.tfvars` with [desired settings](#configuration-settings). Rebuild the container if necessary.
+2. Run `assume-role` to obtain a session.
+3. Run `make apply` to provision kops dependencies with terraform (not the cluster itself)
+4. Run `make kops/shell` to drop into a shell with configured environment for `kops`. Do this any time you want to interact with the cluster.
+5. Run `make kops/build-manifest` to compile the configuration template with current environment settings
+6. Run `make kops/create` to submit the cluster state manifest. Note, no resources will be provisioned.
```
Comment (aknysh) [inline aws/kops/README.md]:
```suggestion
6. Run `make kops/create` to submit the cluster state manifest to the cluster state store. Note, no resources will be provisioned.
```

File: aws/kops/Makefile [hunk 4]
```diff
@@ -13,3 +13,39 @@ apply console destroy graph plan output providers show: init
...
+kops/build-manifest:
+	build-kops-manifest
+
+## Apply the kops manifest
+kops/create:
+	kops create -f $(KOPS_MANIFEST)
+
+## Create the SSH Public Key secret
+kops/create-secret-sshpublickey:
+	kops create secret sshpublickey admin -i $(KOPS_SSH_PUBLIC_KEY_PATH)
```
Comment (joshmyers) [inline aws/kops/Makefile]: Where does `KOPS_SSH_PUBLIC_KEY_PATH` come from? Does there need to be some validation that this is set?

Comment (osterman) [reply aws/kops/Makefile]: Fair question. Documentation on this is [forthcoming](https://github.com/cloudposse/geodesic/pull/339/files#diff-6bfc873b2e29e553a63c1c61498f70ecR46) but lacks this env `KOPS_SSH_PUBLIC_KEY_PATH`.

Literal answer is here: https://github.com/cloudposse/terraform-root-modules/blob/kops-targets/aws/kops/main.tf#L99-L105

**Pull request merged**